app/src/main/java/com/github/swent/echo/compose/components/OfflineUI.kt

```diff
@@ -0,0 +1,104 @@
+package com.github.swent.echo.compose.components
+
+import android.util.Log
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.animateColorAsState
+import androidx.compose.animation.expandHorizontally
+import androidx.compose.animation.shrinkHorizontally
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.material3.CenterAlignedTopAppBar
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.swent.echo.R
+import com.github.swent.echo.connectivity.ConnectionState
+import com.github.swent.echo.connectivity.connectivityState
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
+
+@ExperimentalCoroutinesApi
+@Composable
+fun ConnectivityStatus() {
+    Log.d("ConnectivityStatus", "ConnectivityStatus started")
+    // This will cause re-composition on every network state change
+    val connection by connectivityState()
```

Comment:
Remove `val connection by connectivityState()`, instead add a `connection` parameter. It makes it more modular and will integrate well with the `NetworkService`.